### PR TITLE
[feature](json)support explode_json_object func

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
after this pr we support
expand json object into two column with key column (string) ,and value column(json value)

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return new ExplodeJsonObject(children.get(0));
+    }
+
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        if (!(child().getDataType() instanceof JsonType)) {
+            throw new AnalysisException("only support Json type for explode_json_object function but got "
+                    + child().getDataType());
+        }
+    }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java:64]: i think should check in `checkLegalityAfterRewrite`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    public List<FunctionSignature> getSignatures() {
+        return ImmutableList.of(
+                FunctionSignature.ret(StructLiteral.constructStructType(
+                                ImmutableList.of(StringType.INSTANCE, JsonType.INSTANCE))).args(child().getDataType()));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java:70]: why args is `child().getDataType()` ? it should be JsonType

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObjectOuter.java
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObjectOuter.java:1]: same with ExplodeJsonObject

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java (hunk 2 above)
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java:70]: here child().getDataType() is JsonType?  but here JsonType makes more sense

Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java (hunk 1 above)
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObject.java]: no need to check, done

**Comment by amorynan** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObjectOuter.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public ExplodeJsonObjectOuter withChildren(List<Expression> children) {
+        Preconditions.checkArgument(children.size() == 1);
+        return new ExplodeJsonObjectOuter(children.get(0));
+    }
+
+    @Override
+    public List<FunctionSignature> getSignatures() {
+        return ImmutableList.of(
+                FunctionSignature.ret(StructLiteral.constructStructType(
+                        ImmutableList.of(StringType.INSTANCE, JsonType.INSTANCE))).args(JsonType.INSTANCE));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObjectOuter.java:59]: use static field, refer to explode

Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonObjectOuter.java]: done!

**Comment by amorynan** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: be/src/vec/exprs/table_function/vexplode_json_object.cpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                if (it->value()->isNull()) {
+                    _object_pairs.second->insert_default();
+                } else {
+                    const std::string_view& jsonb_value = std::string_view(
+                            writer->getOutput()->getBuffer(), writer->getOutput()->getSize());
+                    _object_pairs.second->insert_data(jsonb_value.data(), jsonb_value.size());
+                }
+            }
+        }
+        // we do not support other json type except object
```
Comment (eldenmoon) [inline be/src/vec/exprs/table_function/vexplode_json_object.cpp:87]: should we add test cases for other types?

File: be/src/vec/exprs/table_function/vexplode_json_object.cpp [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        ret = assert_cast<ColumnStruct*>(
+                assert_cast<ColumnNullable*>(column.get())->get_nested_column_ptr().get());
+        assert_cast<ColumnUInt8*>(
+                assert_cast<ColumnNullable*>(column.get())->get_null_map_column_ptr().get())
+                ->insert_many_defaults(length);
+    } else if (column->is_column_struct()) {
+        ret = assert_cast<ColumnStruct*>(column.get());
+    } else {
+        throw Exception(ErrorCode::INTERNAL_ERROR,
+                        "only support map column explode to struct column");
```
Comment (eldenmoon) [inline be/src/vec/exprs/table_function/vexplode_json_object.cpp:116]: wrong message?

File: be/src/vec/exprs/table_function/vexplode_json_object.cpp [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    }
+    if (!ret || ret->tuple_size() != 2) {
+        throw Exception(
+                ErrorCode::INTERNAL_ERROR,
+                "only support map column explode to two column, but given:  ", ret->tuple_size());
```
Comment (eldenmoon) [inline be/src/vec/exprs/table_function/vexplode_json_object.cpp:121]: wrong message?

Comment (eldenmoon) [inline be/src/vec/exprs/table_function/vexplode_json_object.cpp:121]: it this message correct？ what's the relationship bettween this function and map type?

Comment (amorynan) [reply be/src/vec/exprs/table_function/vexplode_json_object.cpp:121]: it should be "only support expand json object int to kv pair column, but given:  "

File: be/src/vec/exprs/table_function/vexplode_json_object.cpp (hunk 4 above)
Comment (amorynan) [reply be/src/vec/exprs/table_function/vexplode_json_object.cpp:87]: in qt_order_select_explode_json_object_out case , we treated json other type as NULL for output

File: be/src/vec/exprs/table_function/vexplode_json_object.cpp (hunk 6 above)
Comment (amorynan) [reply be/src/vec/exprs/table_function/vexplode_json_object.cpp:122]: done!

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run feut

**Comment by xiaokang** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by amorynan** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**